### PR TITLE
RFC: cluster: cache GetPlatform() value

### PR DESCRIPTION
**Pull request by ykaliuta**
Jira: https://issues.redhat.com/browse/RHOAIENG-8483

Platform is discovered on start up in main.go and is not supposed to change during program lifetime but it can be time consuming since it performs many api calls.

The patch caches the platform value and returns it if it's set.

It uses nil pointer for the blank value. There is no concurency access problems since when it's stored once from main.go all accesses is read-only.

The patch potentially changes behaviour since now it returns nil error when the actual platform lookup would possible make a error but the actual error is caught in main.go on the first lookup.

Many thanks for submitting your Pull Request ❤️!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md)
- [ ] Pull Request contains description of the issue
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request

## Merge criteria:

- [ ] Have a meaningful commit messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
https://issues.redhat.com/browse/RHOAIENG-8483 can be used as reference for PR

File: pkg/cluster/cluster_config.go [hunk 1]
```diff
@@ -127,6 +129,16 @@ func GetPlatform(cli client.Client) (Platform, error) {
 	return isSelfManaged(cli)
 }
 
+func GetPlatform(cli client.Client) (Platform, error) {
+	if cachedPlatform != nil {
+		return *cachedPlatform, nil
+	}
+
+	platform, err := getPlatform(cli)
```
Comment (zdtsw) [inline pkg/cluster/cluster_config.go:137]: i am reading this https://redhat-internal.slack.com/archives/C064Z5S84F3/p1718199614158759
and start to think if we plan to get this platform write into "release" section.
e.g 
name: Red Hat OpenShift AI(Self-managed)
vs
name: Red Hat OpenShift AI(Managed)

Comment (zdtsw) [reply pkg/cluster/cluster_config.go:137]: How you feel about this?  @VaishnaviHire / @etirelli 

and any later call can check dsci.spec.release.name 
and fall back to call getPlatform()

**Comment by ykaliuta** [comment]
Superseded by  https://github.com/opendatahub-io/opendatahub-operator/pull/1059

**Comment by ykaliuta** [comment]
Superseded by  https://github.com/opendatahub-io/opendatahub-operator/pull/1059

**Pull request closed without merging**